### ISSUE TITLE
Would you interested in employing the sru to speed up inference?

### ISSUE THREAD

**Issue by hwong39** [fatchord/WaveRNN#33]
”SRU is a recurrent unit that can run over 10 times faster than cuDNN LSTM, without loss of accuracy tested on many tasks. “ Sounds very exciting.  I think we can try it. Here is a PyTorch inplement[https://github.com/taolei87/sru](url) and paper details[https://arxiv.org/abs/1709.02755](url).

**Comment by fatchord** [comment]
@hwong39 That looks interesting. I only had a quick look at the paper and it seems like the big speedup is in training because they can do a matrix-vector multiplication with the entire input in a batched operation. This is fine for training when the ground truth is known. However this will not work for inference since each input has to be generated one at a time. 

I only spent two minutes looking at the paper so if I'm missing something please let me know.

**Comment by mrgloom** [comment]
Here is some benchmark:
https://github.com/taolei87/sru/issues/96
